### ISSUE TITLE
Grouping Window Manager sometimes forgets about windows on other workspaces

### ISSUE THREAD

**Issue by ohheyitsjeffrey** [linuxmint/Cinnamon#8258]
```
 * Cinnamon version (4.0.8)
 * Distribution - (Mint 19.1)
 * Graphics hardware *and* driver used
 * 64 bit
 ```

**Issue**
When navigating to other workspaces, sometime open applications dissappear if not pinned, or lost their highlighting and function as if they are not open if pinned.

**Steps to reproduce**
I havent found a consistent way to repoduce but it is happening quite often.  Seems to happen often with Chromium (pinned) and libreoffice, which I think is a seperate issue entirely.

In the video below, you will see as I switch workspaces, Chromium (which is pinned) is sometimes highlighted as open, sometimes not.  Simplescan (not pinned) appears and disappears on certain workspaces.  

![groupwindowbug](https://user-images.githubusercontent.com/16689620/50668089-c9085b80-0f71-11e9-814c-1c8758210c03.gif)

With pinned apps, when opened apps lose their highlighting, they will open a new instance if clicked as if they are not open on another workspace.  Alt-tab still lists the corrects apps.

hopefully this was helpful and please do not hesitate to bug me for more info if necessary.

**Comment by alexisaeff** [comment]
same problem

**Comment by jaszhix** [comment]
Fixed by #8192.
